index.js

```diff
@@ -114,7 +115,7 @@ class HyperDHT extends DHT {
         signature: null
       }
 
-      unann.signature = Persistent.signUnannounce(target, token, from.id, unann, keyPair.secretKey)
+      unann.signature = signUnannounce(target, token, from.id, unann, keyPair.secretKey)
```

Comment:
async also yea? note that this is doable if you just move this logic to a _requestUnnnnounce method or something that is async and call that instead of dht.request below (the unannounces is a list of promises)